### PR TITLE
Replace tzdata Python normalizer with conversion-only flow and update CI/tests

### PR DESCRIPTION
**Pull request by jumpsmm7**
### Motivation

- Remove an in-repo Python package-normalizer that rewrote tzdata archives and instead verify safe conversion and preservation of tar streams, simplifying the CI surface and avoiding content rewrites.
- Align CI workflows and local quality runner with the new conversion-only approach so regressions still run but without exercising the removed Python normalizer or collecting its coverage.

### Description

- Deleted the Python normalizer `tools/normalize-tzdata-package.py` and its unit test `tests/normalize-tzdata-package.py`, and removed references to them from the code-quality matrix and workflows.
- Reworked `tests/update-tzdata-package-info.sh` to validate xz->bzip2 recompression fidelity, ensure package download/publishing expectations, and assert the update script no longer rewrites package contents; updated its output message accordingly.
- Updated `tools/update-tzdata.sh` to preserve the tar stream during recompression and to perform in-place safety checks using an embedded Python snippet rather than invoking the removed normalizer, and to stop rewriting tzdata contents.
- Adjusted CI/workflow and test harness files (`.github/workflows/code-quality-review.yml`, `.github/workflows/shell-validation.yml`, `.github/workflows/update-tzdata.yml`, `sonar-project.properties`, `tools/code-quality.sh`, `tests/code-quality-checks.sh`, and `tests/coderabbit-and-workflow-config-checks.sh`) to run `sh tests/update-tzdata-package-info.sh` as the `tzdata package conversion regression`, removed Python coverage collection, and removed the Python coverage report path from Sonar configuration.

### Testing

- Ran the updated tzdata conversion regression via `sh tests/update-tzdata-package-info.sh` to validate recompression fidelity and update-script expectations, and it passed.
- Executed the shell validation regression matrix (the BusyBox ash checks referenced by `shell-validation.yml`) and the local code-quality checker expectations in `tests/code-quality-checks.sh`, and they passed after updating test expectations to the new conversion test names.
- Verified CI workflow changes to `code-quality-review.yml` and `update-tzdata.yml` and the removal of `sonar.python.coverage.reportPaths` to ensure Sonar/quality gate steps no longer expect the Python coverage artifact.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a8dd97ab1b4832983676aef6db3de7c)

### CODE REVIEW

File: tools/update-tzdata.sh [hunk 1]
```diff
@@ -173,12 +173,6 @@ download_package() {
 		*.zst) zstd --decompress --stdout "${upstream_file}" | bzip2 -9 >"${output_file}" ;;
 	esac
 	tar -tjf "${output_file}" >/dev/null
```
[the remark this answers is not in the document]
Comment (jumpsmm7) [reply tools/update-tzdata.sh:193]: @codex resolve this P1

**Comment by jumpsmm7** [comment]
@codex review the complete PR for final merge readiness, including correctness, security, BusyBox ash compatibility, failure safety, interruption recovery, rollback behavior, permissions, checksums, CI coverage, missing (or broken/failing/non-up-to-date) regression tests, and edge cases

File: tools/update-tzdata.sh [hunk 2]
```diff
@@ -169,31 +169,35 @@ download_package() {
...
 with tarfile.open(archive, "r:bz2") as package:
+    has_posix_timezone = False
     for member in package.getmembers():
         normalized_name = posixpath.normpath(member.name)
         if member.name.startswith("/") or normalized_name == ".." or normalized_name.startswith("../"):
             raise SystemExit(f"Unsafe archive member path: {member.name}")
-        if member.issym() or member.islnk():
+        if (member.isfile() and member.size > 0 and
+                normalized_name.startswith("usr/share/zoneinfo/posix/")):
+            has_posix_timezone = True
```
[the remark this answers is not in the document]
Comment (jumpsmm7) [reply tools/update-tzdata.sh]: @codex resolve this P2

**Comment by jumpsmm7** [comment]
@codex review the complete PR for final merge readiness, including correctness, security, BusyBox ash compatibility, failure safety, interruption recovery, rollback behavior, permissions, checksums, CI coverage, missing (or broken/failing/non-up-to-date) regression tests, and edge cases

File: tools/update-tzdata.sh [hunk 3]
```diff
@@ -169,31 +187,39 @@ download_package() {
...
 
 archive = sys.argv[1]
 with tarfile.open(archive, "r:bz2") as package:
+    has_posix_timezone = False
     for member in package.getmembers():
         normalized_name = posixpath.normpath(member.name)
         if member.name.startswith("/") or normalized_name == ".." or normalized_name.startswith("../"):
             raise SystemExit(f"Unsafe archive member path: {member.name}")
-        if member.issym() or member.islnk():
+        if normalized_name.startswith("usr/share/zoneinfo/posix/") and not member.isdir():
```
[the remark this answers is not in the document]
Comment (jumpsmm7) [reply tools/update-tzdata.sh:206]: @codex resolve this P1

**Comment by jumpsmm7** [comment]
@codex review the complete PR for final merge readiness, including correctness, security, BusyBox ash compatibility, failure safety, interruption recovery, rollback behavior, permissions, checksums, CI coverage, missing (or broken/failing/non-up-to-date) regression tests, and edge cases
